src/Microsoft.AspNetCore.SystemWebAdapters/VirtualPathUtility.cs

```diff
@@ -0,0 +1,702 @@
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Text;
+using System;
+
+namespace System.Web;
+
+/// <summary>
+/// Provides utility methods for common virtual path operations.
+/// </summary>
+/// <remarks>
+/// Portions of this code are based on code originally Copyright (c) 1999 Microsoft Corporation
+/// System.Web.Util.UrlPath code at https://github.com/microsoft/referencesource/blob/master/System.Web/Util/UrlPath.cs
+/// System.Web.Util.StringUtil at https://github.com/microsoft/referencesource/blob/master/System.Web/Util/StringUtil.cs
+/// These files are released under an MIT licence according to https://github.com/microsoft/referencesource#license
+/// </remarks>
+public static class VirtualPathUtility
```

Comment:
Can you add some tests for this?

Reply:
I'll see what I can come up with. There aren't really any good examples of what should become what for these calls that I could find. I did add some exercising code to RequestInfo in the class library example to check it working from both sides and I did some checks of my own to my implementation, but since it relies on HttpRuntime.AppDomainAppVirtualPath for some of its calls, it is harder to test all scenarios.

Reply:
I've added some tests which I think covers most items and scenarios - and indeed caught a couple of things that didn't quite work the same as net4x, which are now fixed.